src/Nethermind/Nethermind.TxPool/Comparison/CompareReplacedTxByFee.cs

```diff
@@ -39,17 +39,12 @@ public int Compare(Transaction? x, Transaction? y)
             if (ReferenceEquals(x, y)) return 0;
             if (ReferenceEquals(null, y)) return 1;
             if (ReferenceEquals(null, x)) return -1;
-            
-            // if gas bottleneck was calculated, it's highest priority for sorting
-            // if not, different method of sorting by gas price is needed
-            if (x.GasBottleneck != 0 || y.GasBottleneck != 0)
-            {
-                y.GasBottleneck.Divide(PartOfFeeRequiredToIncrease, out UInt256 bumpGasBottleneck);
-                return (y.GasBottleneck + bumpGasBottleneck).CompareTo(x.GasBottleneck);
-            }
-            
-            y.GasPrice.Divide(10, out UInt256 bumpGasPrice);
-            return (y.GasPrice + bumpGasPrice).CompareTo(x.GasPrice);
+
+            y.MaxFeePerGas.Divide(PartOfFeeRequiredToIncrease, out UInt256 bumpMaxFeePerGas);
```

Comment:
I see why it is working for Legacy tx, but maybe it will be more clear to write if (tx.IsEip1559) {
}

It will have better readability for my opinion